### PR TITLE
Add Affected to RawApiResponse

### PR DESCRIPTION
**Pull request by sergej-tschak**

### CODE REVIEW

File: src/timetrackings/timeTrackings.test.ts
Comment (benj56) [inline src/timetrackings/timeTrackings.test.ts:79]: As `resultResource` is only used here, let's declare it as `const` here

File: src/timetrackings/timeTrackings.test.ts
Comment (benj56) [inline src/timetrackings/timeTrackings.test.ts:80]: We have an `await` here, so we don't need the `.then` and arrow function, let's just change it to:
```suggestion
    const result = await resultResource;
    expect(...)
```

File: src/utils/response/rawApiResponse.ts
Comment (benj56) [inline src/utils/response/rawApiResponse.ts:3]: This looks just like ApiResponse.ts or response.ts. I think we should think about merging them

File: src/utils/response/rawApiResponse.ts
Comment (benj56) [inline src/utils/response/rawApiResponse.ts:19]: I would consider renaming this to `toRawApiResponse`

**Review by haubitzer — DISMISSED** [review]

File: src/utils/response/rawApiResponse.ts
Comment (benj56) [inline src/utils/response/rawApiResponse.ts:22]: I believe the promise above should be properly typed as `Promise<AxiosResponse<RawApiResponse>>`. I don't believe it should ever resolve to undefined (that would be something we should make sure never happens, in case of an error it needs to reject), therefore this if should be removed

File: src/utils/response/rawApiResponse.ts
Comment (benj56) [inline src/utils/response/rawApiResponse.ts:29]: Let's change this to something more descriptive, like: "The Api response is unsuccessful"

File: src/utils/response/resourceResponse.ts
Comment (benj56) [inline src/utils/response/resourceResponse.ts:4]: I believe this can be a string for some resources such as GeofenceToNode, but not sure

File: src/utils/response/resourceResponse.ts
Comment (benj56) [inline src/utils/response/resourceResponse.ts:8]: Maybe we can directly use the ApiResponse type here instead of using RawApiResponse or ResoureResponse. They all appear to be the same thing to me

**Review by benj56 — CHANGES_REQUESTED** [review]

File: src/utils/response/rawApiResponse.ts
Comment (haubitzer) [reply src/utils/response/rawApiResponse.ts:3]: No, it is not. and it is needed for deltaSync because I new two different types for the results of the deltaSync

File: src/utils/response/rawApiResponse.ts
Comment (haubitzer) [reply src/utils/response/rawApiResponse.ts:22]: promise is never of type Promise<AxiosResponse<RawApiResponse>>

File: src/utils/response/resourceResponse.ts
Comment (haubitzer) [reply src/utils/response/resourceResponse.ts:8]: But they are not.
* ApiResponse is typed with a T 
* RawApiResponse is not Types
* ResourceResponse is typed with a T but doesn't have all fields from ApiResponse.

File: src/utils/response/resourceResponse.ts
Comment (haubitzer) [reply src/utils/response/resourceResponse.ts:4]: Can we check this somehow? I hope it is not the case and if so we don't care at the moment.

**Review by benj56 — DISMISSED** [review]

**Review by AlGvrl — DISMISSED** [review]

**Review by haubitzer — DISMISSED** [review]
Don't forget to bump the version before merging.

File: src/utils/response/rawApiResponse.ts
Comment (tomi-timetac) [inline src/utils/response/rawApiResponse.ts:29]: @whosshell might have more knowledge if this can be removed or not

**Review by tomi-timetac — COMMENTED** [review]
LGTM, @whosshell should approve the removal of Error thrown.

File: src/utils/response/rawApiResponse.ts
Comment (whosshell) [reply src/utils/response/rawApiResponse.ts:29]: We had case before when response was undefined, so since this occurred in the past I would keep some check on this.
And whenever it happens we can cover the case.

Comment (haubitzer) [reply src/utils/response/rawApiResponse.ts:29]: My two cents to this topic, it is not good enough that we only check if the response is undefined and Success is false, we should have a full type guard for the apiResponse to make sure the response looks like what we expect. (https://basarat.gitbook.io/typescript/type-system/typeguard#user-defined-type-guards)

Comment (sergej-tschak) [reply src/utils/response/rawApiResponse.ts]: rolled back error handling

**Review by AlGvrl — APPROVED** [review]

**Review by benj56 — APPROVED** [review]

**Review by tomi-timetac — APPROVED** [review]

**Pull request merged**
